src/main/java/digital/slovensko/autogram/ui/gui/MainMenuController.java

```diff
@@ -110,10 +111,20 @@ private void signFiles(List<File> list) {
         var filesList = getFilesList(list);
         if (filesList.size() == 1) {
             var file = filesList.get(0);
-            var job = SigningJob.buildFromFile(file,
-                    new SaveFileResponder(file, autogram, userSettings.shouldSignPDFAsPades()),
-                    userSettings.isPdfaCompliance(), userSettings.getSignatureLevel(), userSettings.isEn319132(), tspSource, userSettings.isPlainXmlEnabled());
-            autogram.sign(job);
+
+            autogram.wrapInWorkThread(() -> {
```

Comment:
vobes sa mi toto nepozdava imho toto nema vobec vediet tento detail a v else clause sa to nedeje.

Reply:
asi ti úplne nerozumiem